common/persistence/nosql/nosqlplugin/cassandra/gocql/batch.go

```diff
@@ -35,8 +35,8 @@ var _ Batch = (*batch)(nil)
 
 type (
 	batch struct {
-		session *session
-
+		ctx        context.Context
```

Comment:
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx.

https://pkg.go.dev/context

Reply:
But looks like you don't have choice here.

Reply:
Exactly. gocql was retrofitted for context.Context in an unusual way that requires storing the context as a member and we have to work from that interface.